sdks/java/core/src/main/java/org/apache/beam/sdk/function/ThrowingConsumer.java

```diff
@@ -26,6 +26,6 @@
  * interfaces.
  */
 @FunctionalInterface
-public interface ThrowingConsumer<T> {
-  void accept(T t) throws Exception;
+public interface ThrowingConsumer<ExceptionT extends Exception, T> {
+  void accept(T t) throws ExceptionT;
```

Comment:
What is the use case for this change? It looks like you only pass `Exception` as `ExceptionT`. Do you want to define contracts for other subclasses of `Exception`?

Reply:
There was a note at the end of [this comment](https://github.com/apache/beam/pull/7866#issuecomment-465241925) about that part.

It's not used here, but I do have [a motivating case for a `ThrowingConsumer<IOException, T>`](https://github.com/apache/beam/pull/7823/files#diff-1677e5001d798d26293b99815a5c16d8R114) in #7823.